client/CPlayerInterface.cpp

```diff
@@ -900,7 +909,14 @@ void CPlayerInterface::battleEnd(const BattleResult *br)
 
 		if(!battleInt)
 		{
-			GH.pushIntT<BattleResultWindow>(*br, *this);
+			bool replay = allowBattleReplay && !settings["adventure"]["alwaysSkipCombat"].Bool(); //do not allow manual replay
+			allowBattleReplay = false;
+			auto wnd = std::make_shared<BattleResultWindow>(*br, *this, replay);

```

Comment:
Why add a comment when it can be part of the code?
```suggestion
			bool allowManualReplay = allowBattleReplay && !settings["adventure"]["alwaysSkipCombat"].Bool();
			allowBattleReplay = false;
			auto wnd = std::make_shared<BattleResultWindow>(*br, *this, allowManualReplay);

```